core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -142,6 +146,32 @@ public SelfT addContactPoints(@NonNull Collection<InetSocketAddress> contactPoin
    */
   @NonNull
   public SelfT addContactPoint(@NonNull InetSocketAddress contactPoint) {
+    this.programmaticContactPoints.add(new DefaultEndPoint(contactPoint));
+    return self;
+  }
+
+  /**
+   * Adds contact points to use for the initial connection to the cluster.
+   *
+   * <p>You only need this method if you use a custom {@link EndPoint} implementation. Otherwise,
+   * use {@link #addContactPoints(Collection)}.
+   */
+  @NonNull
+  public SelfT addContactEndPoints(@NonNull Collection<EndPoint> contactPoints) {
```

Comment:
Not a huge fan of the name, but I had to change it because it has the same erasure as the `InetSocketAddress` variant.